### PR TITLE
Email reset password

### PR DESCRIPTION
**Pull request by berki0**

### CODE REVIEW

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/EmailService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+@Service
+public class EmailService {
+    private static final Logger logger = LogManager.getLogger(EmailService.class);
+
+    private final JavaMailSender mailSender;
+
+    public EmailService(JavaMailSender mailSender) {
+        this.mailSender = mailSender;
+    }
```
Comment (MirkaaStefanov) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/EmailService.java:18]: Why don't you inject the bean with Lombok annotation AllArgsConstructor

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserService.java [hunk 2]
```diff
@@ -1,39 +1,61 @@
...
+        this.userRepository = userRepository;
+        this.modelMapper = modelMapper;
+        this.jwtService = jwtService;
+        this.tokenService = tokenService;
+        this.passwordResetTokenRepository = passwordResetTokenRepository;
+        this.emailService = emailService;
+        this.emailContentBuilder=emailContentBuilder;
+        this.frontendUrl = frontendUrl;
+    }
+
```
Comment (MirkaaStefanov) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserService.java:58]: For this also I think you can use AllArgsConstructor

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/EmailService.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    public void sendHtmlEmail(String toEmail, String subject, String body) {
+        try {
+            MimeMessage message = mailSender.createMimeMessage();
+            MimeMessageHelper helper = new MimeMessageHelper(message, true, "UTF-8");
+            helper.setFrom("berki07092@gmail.com");
+            helper.setTo(toEmail);
+            helper.setSubject(subject);
+            helper.setText(body, true);
+            mailSender.send(message);
+        } catch (Exception e) {
```
Comment (GeorgiStoyanovAtanasov) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/EmailService.java:28]: Catching generic exceptions is a bad practice, I would change it with } catch (MessagingException e) {
        logger.error("Messaging error occurred while sending email", e);
        throw new CustomEmailException("Messaging error occurred", e);
    } catch (MailException e) {
        logger.error("Mail server error occurred", e);
        throw new CustomEmailException("Mail server error occurred", e);
    }

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/EmailService.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    private final JavaMailSender mailSender;
+
+    public EmailService(JavaMailSender mailSender) {
+        this.mailSender = mailSender;
+    }
+    public void sendHtmlEmail(String toEmail, String subject, String body) {
+        try {
+            MimeMessage message = mailSender.createMimeMessage();
+            MimeMessageHelper helper = new MimeMessageHelper(message, true, "UTF-8");
+            helper.setFrom("berki07092@gmail.com");
```
Comment (berki0) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/EmailService.java:23]: can you get the value of email form application.property

**Pull request merged**